src/dbnode/topology/map.go

```diff
@@ -69,8 +72,24 @@ func NewStaticMap(opts StaticOptions) Map {
 				append(topoMap.orderedShardHostsByShard[id], elem)
 		}
 	}
+	for _, hostShardSet := range hostShardSets {
+		host := hostShardSet.Host()
 
-	return &topoMap
+		for _, shard := range hostShardSet.ShardSet().All() {
+			if shard.SourceID() != "" {
```

Comment:
Is SourceID set for all shard replicas in our clusters? my guess is it's only used during replaces but can you confirm?

Reply:
is it retained after that?